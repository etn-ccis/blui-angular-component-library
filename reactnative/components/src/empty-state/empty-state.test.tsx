import React from 'react';
import renderer from 'react-test-renderer';
import { Button, Icon as RNIcon } from 'react-native-elements';
import { wrapIcon, EmptyState } from '..';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
const Line = wrapIcon({ IconClass: Icon, name: 'chart-line-variant' });


describe('EmptyState Tests ', function () {

   it('Icon Renders', () => {
      const tree = renderer.create(
         <EmptyState title={'Test'} IconClass={Line}/>
      ).toJSON();
      expect(tree).toMatchSnapshot();
   });

   it('Only Title Renders', () => {
      const tree = renderer.create(
         <EmptyState title={'EmptyState'} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
   });

   it('Description Renders', () => {
      const tree = renderer.create(
         <EmptyState title={'Test'} description={'Description'} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
   });

   it('Actions Renders', () => {
      const tree = renderer.create(
         <EmptyState title={'Test'} actions={
            <Button
               icon={<RNIcon name="add-circle-outline"/>}
               title=" Add Device"
            />
         } />
      ).toJSON();
      expect(tree).toMatchSnapshot();
   });

   it('Title, actions, Description Renders', () => {
      const tree = renderer.create(
         <EmptyState  title={'EmptyState'} description={'Description'} actions={
            <Button
               icon={<RNIcon name="add-circle-outline"/>}
               title=" Add Device"
            />
         }/>
      ).toJSON();
      expect(tree).toMatchSnapshot();
   });
});

