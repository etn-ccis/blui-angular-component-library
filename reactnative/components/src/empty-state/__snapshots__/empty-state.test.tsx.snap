// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState Tests  Actions Renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
        },
        Object {
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "600",
          "letterSpacing": 0,
        },
        Object {},
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    Test
  </Text>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
              underlayColor="white"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontSize": 24,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
             Add Device
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`EmptyState Tests  Description Renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
        },
        Object {
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "600",
          "letterSpacing": 0,
        },
        Object {},
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    Test
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#6200EE",
        },
        Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "600",
        },
        Object {},
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    Description
  </Text>
</View>
`;

exports[`EmptyState Tests  Icon Renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 100,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="icon"
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
        },
        Object {
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "600",
          "letterSpacing": 0,
        },
        Object {},
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`EmptyState Tests  Only Title Renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
        },
        Object {
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "600",
          "letterSpacing": 0,
        },
        Object {},
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    EmptyState
  </Text>
</View>
`;

exports[`EmptyState Tests  Title, actions, Description Renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
        },
        Object {
          "fontFamily": "System",
          "fontSize": 20,
          "fontWeight": "600",
          "letterSpacing": 0,
        },
        Object {},
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    EmptyState
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#6200EE",
        },
        Object {
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "600",
        },
        Object {},
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    Description
  </Text>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
              underlayColor="white"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontSize": 24,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
             Add Device
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
